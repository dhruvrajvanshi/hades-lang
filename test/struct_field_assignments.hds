extern fn puts(*cchar): Void = puts

fn main(): Void {
    val mut box = Box(c"initial")
    val mut box_ptr = &mut box
    val field_ptr = &mut box_ptr.value
    puts(*field_ptr)
    box_ptr.value = c"change 1"

    puts(box_ptr.value)

    *field_ptr = c"change 2"

    puts(box_ptr.value)

}

struct Box[T] {
    val value: T
}
