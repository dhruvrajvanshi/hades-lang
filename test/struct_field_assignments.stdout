initial
change 1
change 2